Client/Frontend/Home/ActivityStreamTopSitesCell.swift

```diff
@@ -218,14 +218,18 @@ class HorizontalFlowLayout: UICollectionViewLayout {
     private var boundsSize = CGSize.zero
     private var insets = UIEdgeInsetsZero
     private let minimumInsets: CGFloat = 20
-    var numberOfPages = 0
 
     override func prepareLayout() {
         super.prepareLayout()
         cellCount = self.collectionView!.numberOfItemsInSection(0)
         boundsSize = self.collectionView!.bounds.size
     }
 
+    func numberOfPages() -> Int {
+        let itemsPerPage = maxVerticalItemsCount() * maxHorizontalItemsCount()
+        return itemsPerPage == 0 ? 1 : Int(ceil(Double(cellCount) / Double(itemsPerPage)))
```

Comment:
You could return 0. Doesnt matter I dont think. the real issue is when `itemsPerPage` is 0 it tries to divide by 0.